src/backend/base/langflow/components/models/AIMLModel.py

```diff
@@ -71,24 +64,33 @@ class AIMLModelComponent(LCModelComponent):
         ),
     ]
 
+    def update_build_config(self, field_value: str, build_config: dict, field_name: str | None = None):
+        if field_name == "api_key" or field_name == "aiml_api_base" or field_name == "model_name":
+            aiml = AimlModels()
+            aiml.get_aiml_models()
```

Comment:
This is going to make an API call to the model endpoint each time this is called, which can be very frequent. I would add a basic caching mechanism in your aiml_constants class that retrieves the models that you store in `self.chat_models`, if already populated.

Reply:
The function is called only once when the user places the AI/ML component on the desk. So, I don’t think we need to use caching or anything similar, as a single request works quickly.

Reply:
Ah, you are correct. The `update_build_config` method is only called dynamically if `real_time_refresh` flag is set on a field (and agreed that you do not need that right now). 

Though, you might want that later on, in order to support dynamic inputs for different models.